tsfile/src/main/java/org/apache/iotdb/tsfile/compress/ICompressor.java

```diff
@@ -311,4 +316,61 @@ public CompressionType getType() {
       return GZIP;
     }
   }
+
+  class ZstdCompressor implements ICompressor {
+
+    private int compressionLevel;
+
+    public ZstdCompressor() {
+      super();
+      compressionLevel = Zstd.maxCompressionLevel();
+    }
+
+    @Override
+    public byte[] compress(byte[] data) throws IOException {
+      return Zstd.compress(data, compressionLevel);
+    }
+
+    @Override
+    public byte[] compress(byte[] data, int offset, int length) throws IOException {
+      if (data == null) {
+        return new byte[0];
+      }
+      byte[] toCompressData;
+      if (offset == 0 && length == data.length) {
+        toCompressData = data;
+      } else {
+        toCompressData = new byte[length];
+        System.arraycopy(data, offset, toCompressData, 0, length);
```

Comment:
I'm not sure. Is this necessary?

Reply:
another implementation
![image](https://user-images.githubusercontent.com/55970239/223663789-456ce631-3781-48f8-9aaa-86dbb25c107b.png)